WIP: Initial Concept for Armory Restructure

shenshaw26 opened the pull request:
@mwartell , I know you are working on this...just was bothering me last night so I threw something together for discussion.  Lets talk about this a bit later

Commit:
initial hack at pydantic experiment

Commit:
initial hack at pydantic experiment

Commit:
Initial stab at environment

Commit:
Adding quick check

Commit:
minor update

mwartell commented on armory/environment.py:
```diff
@@ -2,4 +2,243 @@
+    necessary for armory execution.
+    """
+
+    ARMORY_CONFIGURATION_DIRECTORY: str = os.path.expanduser("~")
```
because these objects only exist in an EnvironmentConfiguration, the `ARMORY_` part is redundant and adds verbosity without adding information

mwartell replied:
And ALL_CAPS is tolerable in global (module-scope) variables, but wrong as a member name.

It also hurts my eyes and brain.

Commit:
Saving State

Commit:
Saving State

Commit:
Merging primo into current

Commit:
Adding todos

Commit:
saving state

Commit:
saving state

Commit:
Got Env setup going

Commit:
cleanup

Commit:
Getting Basic Experiment Idea fleshed out

Commit:
saving state

Commit:
saving state

davidslater committed:
move ucf101 clipping to data (#1309)

* move ucf101 clipping to data

* update deprecation for better searching

* remove print

Commit:
Merge branch 'duo' into update_armory

shenshaw26 committed:
duo armory improvements (#1307)

* Just getting started

* example for logging update

* adopt loguru

this only changes main.py
other modules still use logging
you must use `--no-docker` with this change, until they get loguru

* add duration to log messages
add function to add new sinks

* initial no-docker bits setup

* adding no-docker runs to pytest suite

* removing interactive use case

* Updating ci file to run on demand

* messing with ci

* updating docs

* Adding interactive tests back

* removing test_logs example

* Cleaning up flake config

* Adding more verbosity on git pre-commit hooks

* more docs update

* full excision of all logging and colored logs use

* rollback version to develop to unbreak CI

* review and flake8 corrections

* updating docs and renaming requirements file and fixed flake issue

* log/logger name collision cleared

* ci uses new no-docker-requirement.txt

* Removing unnecessary req, see comments from https://github.com/twosixlabs/armory/pull/1271#pullrequestreview-878805272

* experiment pin numpy version as in tf1/Dockerfile

the log from CI shows an upgrade/downgrade for numpy
ending with 1.19.5
this may be the cause of the CI fault

* Revert "experiment pin numpy version as in tf1/Dockerfile"

This reverts commit 37e81a2ce9cb6d1d96e00776fd624ac93582a08b.

* Addressing PR1271 comments

* add loguru install to dockerfiles

* Addressing PR1271 comments

* Addressing PR1271 comments

* Addressing PR1271 comments

* More doc updates for PR

* intercept and log other libraries calls to logging

* logs documentation

* Revert "add loguru install to dockerfiles"

the 4c058b99 commit is correlated to the second with a CI failure in run #2227
this is an experiment to see if it unbreaks CI; if it does, the addition
of loguru to the dockerfiles still needs to be done.

This reverts commit 4c058b99a1b11b119c0a520cbea1c0b5c7d0b1d6.

* minor updates to comments / docs

* added back in loguru, but pip not conda

* add loguru to pip install for all dockerfiles

* could it be as easy as pinning numpy version

as is done in tf1

* Revert "could it be as easy as pinning numpy version"

No, it had no effect. Armory CI #2239 failed with the same error.

This reverts commit 5f87c0d5d3f06b1410e9d0f537e0d696d57c3406.

* fix tf2 ci test

* first cut logging docs

* fix argument error

* fixing minor doc comment

* broken commit - moving to unified log-level

commiting here because there are too many moving pieces

* bump release version

* Revert "broken commit - moving to unified log-level"

This reverts commit eef41aaaa268841aee404172306e1e6e16e5b0cc.

* calm messages from chatty modules

* bump version in scenarios

* broken - logging filters start

* broken - added --log-level option

I'm now getting a runpy.py error on the launch of the subordinate
python instance at
armory.scenarios.main

* add callchain doc

* getting CI setup

* logging function decorator

* third try

* Getting Docker Square... no longer going to support tf1 or deepspeech

* pass filter settings to subordinate python

* address all pr #1271 comments

* Updating tests to get datasets working

* fix comment typo

* flake8 ci unref import fix

* small refactor to make more general

* flake8 undef name fix

* First stab at models

* added more model tests

* added more model tests

* first draft configuration object

* fix comment typo

* removed global config in favor of actual argument

* updating datasets test and adding to CI

* Initial hack at test_models.py

* updating notes

* Cleaning up test_host

* adding notes

* Cleaning up test_docker

* Cleaning up test_docker

* Refactor test_host/test_external_repo and tests/test_docker/test_weight_download to single file test_download

* Updating tests/test_datasets.py to include split_index

* cleaning up dataset tests

* moving test_metrics to top level

* cleanup

* more simple refactor

* merging in relevant bits from slater tf1 deprecation branch

* Test refactor complete...first attempt at getting CI to work

* fixing tests and docs

* fixing up docker file

* another error

* gotta build base for now

* launcher app spec

* use log name for uniformity

* fixing tests and ci

* more app docs

* another try

* another try

* add progress log level

* updating tools

* one more

* one more

* Cleanup from merge and adding pytorch-deepspeech docker

* making progress

* minor log tweaks based on real runs

* make docker error message match build.sh

* made the logging planning doc current

* minor cleanup to get tests better

* new thing

* fixing pre-commit crap

* I think its working

* pytest spec in ci_test.yml was wrong...correcting

* one last fix on docker side

* minor update to build script documentation

* initial hack at manual container push

* trying from github

* removing locally

* minor tweak

* fix workflow

* Initial start...not working yet but close

* saving state

* minor update

* saving state

* Initial setup working...now to see if all tests work

* Removing explicit armory version deps in scenario configs

* Updating Test to make it pass (no longer makes sense to do that check

* Adding some helpful log messages

* trying to pip install in test_configuration bit

* addressing review comments

* Fixing build tag from armory version

* Addressing PR comments

* adding docs for pr

* adding armory configure to ci tests

* another attempt at armory configure

* another try

* another try

* I hate broken pipes

* I hate broken pipes

* gave up just added to armory configure '--use-defaults' flag

* Adding back all tests

* this one is for all the marbles

* Fixing test to work with new version

* fix broken default

* this makes sense to me

* WIP: Update Armory to use `setuptools_scm` (#8)

* third try

* Getting Docker Square... no longer going to support tf1 or deepspeech

* Updating tests to get datasets working

* small refactor to make more general

* First stab at models

* added more model tests

* added more model tests

* updating datasets test and adding to CI

* Initial hack at test_models.py

* updating notes

* Cleaning up test_host

* adding notes

* Cleaning up test_docker

* Cleaning up test_docker

* Refactor test_host/test_external_repo and tests/test_docker/test_weight_download to single file test_download

* Updating tests/test_datasets.py to include split_index

* cleaning up dataset tests

* moving test_metrics to top level

* cleanup

* more simple refactor

* merging in relevant bits from slater tf1 deprecation branch

* Test refactor complete...first attempt at getting CI to work

* fixing tests and docs

* fixing up docker file

* another error

* gotta build base for now

* fixing tests and ci

* another try

* another try

* updating tools

* one more

* one more

* Simple docker pytorch deepspeech (#1293)

* cpu problems

* update all librispeech configs

* fix dockerfile

* update deepspeech-pytorch

* remove tf1 from configs and enable compatibility on tf2 (#1290)

* bump patch version (#1299)

the 0.14.6 represents the next release that we might
make, not the 0.14.5 source-only released

* Cleanup from merge and adding pytorch-deepspeech docker

* making progress

* make docker error message match build.sh

* minor cleanup to get tests better

* new thing

* fixing pre-commit crap

* I think its working

* pytest spec in ci_test.yml was wrong...correcting

* one last fix on docker side

* minor update to build script documentation

* initial hack at manual container push

* trying from github

* removing locally

* minor tweak

* fix workflow

* Initial start...not working yet but close

* saving state

* minor update

* saving state

* Initial setup working...now to see if all tests work

* Removing explicit armory version deps in scenario configs

* Updating Test to make it pass (no longer makes sense to do that check

* Adding some helpful log messages

* trying to pip install in test_configuration bit

* addressing review comments

* Fixing build tag from armory version

* Addressing PR comments

* adding docs for pr

* adding armory configure to ci tests

* another attempt at armory configure

* another try

* another try

* I hate broken pipes

* I hate broken pipes

* gave up just added to armory configure '--use-defaults' flag

* Adding back all tests

* this one is for all the marbles

* Fixing test to work with new version

* fix broken default

* this makes sense to me

Co-authored-by: Seth Henshaw <seth.henshaw@twosixtech.com>
Co-authored-by: davidslater <david.slater@twosixlabs.com>
Co-authored-by: matt wartell <matt.wartell@twosixlabs.com>
Co-authored-by: matt wartell <matt.wartell@twosixtech.com>

* Updating small bits and skipping docker_build test when inside docker

* minor format

* put container id in shutdown message

* scrap the interface prototype

* Trying to Get the merge to have the right bits

* a helpful check when testing optional cuda support

* duo is the new primo in CI

* fix a flake8 report
                                and re-add my git commit hook

* adding end2end cron

* testing

* try again

* 3rd try

* adding needs job

* adding needs job no build

* fixing up

* another try:

* Trying different tag mechanism

* just trying passin value

* another try

* I think its working

* fixing pytest to use markers (-m) properly

* Cleaning up

* another shot

* fixing pycache issue

* fixing docker ci tests

* fixing weird bug

* last try for the day

* Adding note

* making runner logs more verbose

* trying single run on end2end

* trying single run on end2end

* just trying unit tests in docker on local runner

* change base level to warning

* Putting everything back to try with local runner on noether

* try again

* clamp progress unless requested

* supress progress bar unless is_progress

* pytest should only capture ERROR logs

* use scm generated tag for docker images

* add scm tags to docker builder

* demote log config messages to trace

* prefer base:latest for all flavor builds

@shenshaw26 should check that this is the best way

* argparse accepts =; removed base build

* fix dry-run handling

* remove duplicate art install

* change dry-run boolean to string

* framework argument is now freely positioned

* prevent --debug modifying armory level

* remove tf1 references

* use short scm strings everywhere

* use local versioned image, unversioned, then pull

* don't append image tag if already present

* not yet working build.sh replacement

* couldn't get dockerapi to work use shell

* use explicit build directory

* update init

* update CI workflow

* add f to string

* update CI docker build

* update help to remove tf1

* pip install before building docker containers

* typo

* add deprecration warning

* reorder args for better help ordering

* update help and remove wrong comments

* update name reference

* update references from build.sh to build.py

* remove 'dev' tags

* remove version updaters

* format

* yamllint

* removed --tag since it wasn't used or implemented

* remove unused import

Co-authored-by: Seth Henshaw <seth.henshaw@twosixtech.com>
Co-authored-by: matt wartell <matt.wartell@twosixlabs.com>
Co-authored-by: David Slater <david.slater@twosixlabs.com>
Co-authored-by: matt wartell <matt.wartell@twosixtech.com>

davidslater committed:
logger to log (#1322)

Commit:
run tests on github not self-hosted

the self-hosted was an experiment on duo that we are
not using (yet) on develop

Commit:
remove branch duo ci triggers

Commit:
merging tiga -> update_armory

Commit:
adding CI bits for tiga

Commit:
making it a todo

Commit:
prefer native importlib.metadata if available (#1327)

davidslater committed:
"`get` no longer imported from `armory.scenarios.main` in `armory.scenarios.__init__" (#1312)

Commit:
enable log.exception backtraces (#1331)

davidslater committed:
Tfds v4, ART v1.10, and other docker/build updates  (#1323)

* tfds 3->4

* no ART

* update librispeech version

* update dockerfiles

* remove version

* update build scripts to add all and base

* update to fix librispeech for tfds v4

* update tests and release

* revert docker image to previous release

* add TODO

* don't rebuild base in build test

* remove commented old code

Commit:
Merge branch 'develop' into tiga

davidslater committed:
update requirements (#1337)

* update requirements

* build requirements

davidslater committed:
add checks for evaluate_current (#1338)

davidslater committed:
Fix GPU usage with PyTorch / TF (#1341)

* move to __init__

* remove unneeded line

davidslater committed:
Update - Externally downloaded repos (requirements in armory repo but not in docker image) (#1325)

* add friendly error messages for external downloaded repos

* minor update

* import style

* no logger needed

* update doc

* move/rename errors.py

davidslater committed:
remove mongoDB functionality (#1346)

davidslater committed:
provide consistent ART data pathing (#1352)

* provide consistent ART data pathing

* import except

* remove import

* fix race condition on pathing

davidslater committed:
Improve Docker Image Lookup (#1344)

* move to images

* update images

* move docker_api to docker

* update docker API use

* update naming

* refactor get local

davidslater committed:
update torch loader (#1359)

davidslater committed:
remove armory clean (#1361)

davidslater committed:
already should be removed (#1364)

davidslater committed:
warning (#1362)

davidslater committed:
update name in test (#1365)

davidslater committed:
update scenario to properly override Scenario (#1363)

* update scenario to properly override

* add task_kwargs

Commit:
Merge remote-tracking branch 'origin/develop' into tiga

Commit:
Merge branch 'tiga' into update_armory
